core/flow/rule.go

```diff
@@ -110,10 +120,19 @@ func (r *Rule) isEqualsTo(newRule *Rule) bool {
 	}
 	if !(r.Resource == newRule.Resource && r.RelationStrategy == newRule.RelationStrategy &&
 		r.RefResource == newRule.RefResource && r.StatIntervalInMs == newRule.StatIntervalInMs &&
-		r.TokenCalculateStrategy == newRule.TokenCalculateStrategy && r.ControlBehavior == newRule.ControlBehavior && util.Float64Equals(r.Threshold, newRule.Threshold) &&
-		r.MaxQueueingTimeMs == newRule.MaxQueueingTimeMs && r.WarmUpPeriodSec == newRule.WarmUpPeriodSec && r.WarmUpColdFactor == newRule.WarmUpColdFactor) {
+
+		r.TokenCalculateStrategy == newRule.TokenCalculateStrategy && r.ControlBehavior == newRule.ControlBehavior &&
+		util.Float64Equals(r.Threshold, newRule.Threshold) &&
+
+		r.MaxQueueingTimeMs == newRule.MaxQueueingTimeMs && r.WarmUpPeriodSec == newRule.WarmUpPeriodSec &&
+		r.WarmUpColdFactor == newRule.WarmUpColdFactor &&
+
+		r.SafeThreshold == newRule.SafeThreshold && r.RiskThreshold == newRule.RiskThreshold &&
+		r.LowWaterMark == newRule.LowWaterMark && r.HighWaterMark == newRule.HighWaterMark) {
```

Comment:
Direct token calculate strategy and throttling control behavior don't use stat,  needStatistic() function should be modified.

Reply:
updated